### PR TITLE
Button: Adding transparent variant

### LINKED ISSUE #15105: Update ghost variant css once design is finalized
Daisy is working on finalizing the design for the button ghost variant. This issue tracks the followups needed after finalized design

### PR DESCRIPTION
**Pull request by khmakoto**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #15105
- [x] Include a change request file using `$ yarn change`

#### Description of changes

This PR adds the `transparent` variant button that imitates the styling of the `ActionButton` in v7. Examples and visual regression tests have been updated to reflect this.

#### Focus areas to test

(optional)

### CODE REVIEW

**Review by Ridvanovskyy — APPROVED** [review]

File: packages/react-button/src/components/Button/__snapshots__/Button.test.tsx.snap [hunk 1]
```diff
@@ -251,8 +247,8 @@ exports[`Button renders anchor when href prop is provided 1`] = `
         --color-brand-borderColor: transparent;
         --color-brand-checked-background: #005a9e;
         --color-brand-checked-contentColor: #ffffff;
-        --color-brand-checkedHovered-background: #005a9e;
-        --color-brand-checkedHovered-contentColor: #ffffff;
+        --color-brand-checked-hovered-background: #005a9e;
```
Comment (dzearing) [inline packages/react-button/src/components/Button/__snapshots__/Button.test.tsx.snap:250]: this does not seem right. I think the states should all be at the same level.

File: packages/react-button/src/components/ToggleButton/useToggleButtonClasses.tsx [hunk 2]
```diff
@@ -10,48 +10,85 @@ export const useToggleButtonClasses = makeVariantClasses<ToggleButtonState>({
...
-      checkedPressed: {
-        background: 'var(--ghost-checkedPressed-background)',
-        contentColor: 'var(--ghost-checkedPressed-contentColor)',
+
+      transparent: {
+        checked: {
+          background: 'transparent',
+          contentColor: palette?.themePrimary,
+          iconColor: palette?.themePrimary,
+          hovered: {
```
Comment (dzearing) [inline packages/react-button/src/components/ToggleButton/useToggleButtonClasses.tsx:85]: this should be collapsed as `checkedHovered`.

Comment (dzearing) [reply packages/react-button/src/components/ToggleButton/useToggleButtonClasses.tsx:85]: See `ColorTokens` in Theme. This should really be complaining in the typings. Can you see if you can make this typesafe?

File: packages/react-button/src/components/ToggleButton/useToggleButtonClasses.tsx [hunk 3]
```diff
@@ -10,48 +10,85 @@ export const useToggleButtonClasses = makeVariantClasses<ToggleButtonState>({
...
+
+      ghost: {
+        checked: {
+          background: palette?.neutralLight,
+          contentColor: palette?.black,
+          iconColor: palette?.themeDarker,
+
+          hovered: {
+            background: palette?.neutralQuaternaryAlt,
+            contentColor: palette?.themePrimary,
```
Comment (xugao) [inline packages/react-button/src/components/ToggleButton/useToggleButtonClasses.tsx:74]: could you also update the ghost to align with `CommandBarButton` styling? so in this case, we should keep the contentColor as is when hovered. i think this is the only change we need for `ghost`?

File: apps/vr-tests/src/stories/ButtonNext.stories.tsx [hunk 4]
```diff
@@ -131,8 +142,13 @@ storiesOf('Button Next - With icon before content', module)
...
     </Button>
   ))
-  .addStory('Primary Disabled', () => (
-    <Button primary disabled icon="X">
+  .addStory('Transparent', () => (
+    <Button transparent icon="X">
+      Hello, world
+    </Button>
+  ))
+  .addStory('Transparent Disabled', () => (
```
Comment (xugao) [inline apps/vr-tests/src/stories/ButtonNext.stories.tsx:150]: let's also add a test for both `transparent` and `ghost` are set (which is what user needs to do if they need `ActionButton`)

File: packages/react-button/src/components/ToggleButton/useToggleButtonClasses.tsx (hunk 2 above)
Comment (khmakoto) [reply packages/react-button/src/components/ToggleButton/useToggleButtonClasses.tsx:85]: Spoke with @dzearing offline, I will address collapsing things in this PR but I'll check the typings in a different PR.

File: packages/react-button/src/components/Button/__snapshots__/Button.test.tsx.snap (hunk 1 above)
Comment (khmakoto) [reply packages/react-button/src/components/Button/__snapshots__/Button.test.tsx.snap]: Fixed!

File: packages/react-button/src/components/ToggleButton/useToggleButtonClasses.tsx (hunk 3 above)
Comment (khmakoto) [reply packages/react-button/src/components/ToggleButton/useToggleButtonClasses.tsx]: Done!

**Review by xugao — APPROVED** [review]

**Pull request merged**